torch/csrc/jit/python/init.cpp

```diff
@@ -620,8 +620,9 @@ void initJITBindings(PyObject* module) {
       .def(
           "_jit_set_profiling_mode",
           [](bool profiling_flag) {
-            bool oldState = getProfilingMode();
-            getProfilingMode() = profiling_flag;
+            TORCH_WARN("Set profiling mode is deprecated and will invoke getGraphExecutorOptimize now. Please use `_get_graph_executor_optimize`");
```

Comment:
Should we also do this for the c++ implementation of getProfilingMode() in `profiling_graph_executor_impl.cpp` ? e.g. there are some internal users of torch::jit::getProfilingMode() that might break with this change